### PR TITLE
chore: misc clean up in `forest_crypto`

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- [x] `verify_bls_sig` is the same as `fvm_shared::crypto::signature::ops::verify_bls_sig`.
- [x] All JSON serialization helpers should be moved to `forest_json`.
- [x] There's no need to distinguish the error enum. It can be replace with `anyhow::Error`.
- [x] The `Signer` trait should be deleted. It is only used for testing with `DummySigner`. `SignedMessage` will require a new method: `new_unchecked(message: Message, signature: Signature) -> SignedMessage`. It is similar to `new_from_parts` but doesn't verify the signature. All `DummySigner`s should be removed.
- [x] Merge the duplicate JSON serializers for `Ticket` and `VRFProof`.

**Reference issue to close (if applicable)**

Closes  https://github.com/ChainSafe/forest/issues/2113

**Other information and links**

### CODE REVIEW

File: utils/test_utils/src/chain_structures.rs [hunk 1]
```diff
@@ -126,13 +128,20 @@ pub fn construct_full_tipset() -> FullTipset {
...
+impl DummySigner {
+    pub fn sign_bytes(_: &[u8], _: &Address) -> Result<Signature> {
         Ok(Signature::new_secp256k1(DUMMY_SIG.to_vec()))
     }
+
+    pub fn sign_message(message: Message) -> Result<SignedMessage> {
+        let bz = message.cid()?.to_bytes();
+        let signature = Self::sign_bytes(&bz, &message.from)?;
+        Ok(SignedMessage::new_unchecked(message, signature))
+    }
```
Comment (lemmih) [inline utils/test_utils/src/chain_structures.rs:144]: Creating a fake signature can never fail.

```suggestion
    pub fn sign_message(message: Message) -> SignedMessage {
        let signature = Signature::new_secp256k1(DUMMY_SIG.to_vec());
        SignedMessage::new_unchecked(message, signature)
    }
```

File: utils/test_utils/src/chain_structures.rs [hunk 2]
```diff
@@ -126,13 +128,20 @@ pub fn construct_full_tipset() -> FullTipset {
     FullTipset::new(blocks).unwrap()
 }
 
-const DUMMY_SIG: [u8; 1] = [0u8];
+pub const DUMMY_SIG: [u8; 1] = [0u8];
```
Comment (lemmih) [inline utils/test_utils/src/chain_structures.rs:131]:
```suggestion
const DUMMY_SIG: [u8; 1] = [0u8];
```

File: vm/message/src/signed_message.rs [hunk 3]
```diff
@@ -19,21 +18,19 @@ pub struct SignedMessage {
...
-
-        let signature = signer
-            .sign_bytes(&bz, &message.from)
-            .map_err(|e| CryptoError::SigningError(e.to_string()))?;
-
+    /// Generate a new signed message from fields.
+    /// The signature will be verified.
+    pub fn new_from_parts(message: Message, signature: Signature) -> anyhow::Result<SignedMessage> {
+        signature
+            .verify(&message.cid().unwrap().to_bytes(), &message.from)
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:25]:
```suggestion
            .verify(&message.cid()?.to_bytes(), &message.from)
```

File: vm/message/src/signed_message.rs [hunk 4]
```diff
@@ -220,27 +217,17 @@ pub mod json {
...
-
-            SignedMessage::new(
-                crate::message::tests::MessageWrapper::arbitrary(g).message,
-                &DummySigner,
-            )
-            .unwrap()
+            let message = crate::message::tests::MessageWrapper::arbitrary(g).message;
+            let bz = message.cid().unwrap().to_bytes();
+            let signature = DummySigner::sign_bytes(&bz, &message.from).unwrap();
+            SignedMessage::new_unchecked(message, signature)
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:230]:
```suggestion
            let message = crate::message::tests::MessageWrapper::arbitrary(g).message;
            DummySigner::sign_message(message)
```

**Review by lemmih — CHANGES_REQUESTED** [review]

**Comment by lemmih** [comment]
I feel it would be better to get rid of `DummySigner` altogether. You can manually create a dummy signature in the tests (and even generate an arbitrary one in the quickchecks).

File: utils/test_utils/src/chain_structures.rs (hunk 2 above)
Comment (hanabi1224) [reply utils/test_utils/src/chain_structures.rs]: Have removed `DUMMY_SIG` together with `DummySigner`

File: vm/message/src/signed_message.rs (hunk 3 above)
Comment (hanabi1224) [reply vm/message/src/signed_message.rs]: Have removed `DummySigner`

File: vm/message/src/signed_message.rs (hunk 4 above)
Comment (hanabi1224) [reply vm/message/src/signed_message.rs]: Have removed `DummySigner`

File: utils/test_utils/src/chain_structures.rs (hunk 1 above)
Comment (hanabi1224) [reply utils/test_utils/src/chain_structures.rs]: Have removed `DummySigner`

File: blockchain/message_pool/src/msgpool/utils.rs [hunk 5]
```diff
@@ -47,6 +47,7 @@ pub(crate) async fn recover_sig(
     let val = bls_sig_cache
         .get(&msg.cid()?)
         .ok_or_else(|| Error::Other("Could not recover sig".to_owned()))?;
-    let smsg = SignedMessage::new_from_parts(msg, val.clone()).map_err(Error::Other)?;
+    let smsg = SignedMessage::new_from_parts(msg, val.clone())
+        .map_err(|err| Error::Other(err.to_string()))?;
```
Comment (lemmih) [inline blockchain/message_pool/src/msgpool/utils.rs:51]: We already have this:
```rust
impl From<anyhow::Error> for Error {
    fn from(e: anyhow::Error) -> Self {
        Error::Other(e.to_string())
    }
}
```

```suggestion
    let smsg = SignedMessage::new_from_parts(msg, val.clone())?;
```

**Review by lemmih — APPROVED** [review]
Looking good.

File: blockchain/message_pool/src/msgpool/utils.rs (hunk 5 above)
Comment (hanabi1224) [reply blockchain/message_pool/src/msgpool/utils.rs]: fixed

**Pull request merged**